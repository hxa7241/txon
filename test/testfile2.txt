
DC.title:`TXON: Text Object Notation -- Textual Punctuation For Data`
=====================================================================

*** unfinished ***


DC:`
   creator:`Harrison Ainsworth / HXA7241`

   description:`Definition of a minimal general data format that you can write straight into any text.`
   subject:`data format, text, JSON, software tool, python`
   type:`technical article`
   relation:`http://www.hxa.name/txon/`
   language:`en-GB`

   date:`2010-04-02`
   date:`2010-05-18`
`




Definition
----------

It is a nestable key-value syntax structure: like a simplified JSON with only strings and objects, and a different textual representation. The basic structure is:

   name:`value`

This can be nested:

   name:` name2:`nested value` `

And names are optional and need not be unique, text content may be included alongside terms, and any layout is allowed:

   references:`
      JSON -- :`http://json.org/`
      Lua --  :`http://lua.org/`
   `

Outside an expression, only the sequence of : followed by ` is special and not generally allowed. Inside an expression, ` is special and not generally allowed, unless escaped with \.

