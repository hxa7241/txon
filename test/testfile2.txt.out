{
  "DC":
  {
    "creator":"Harrison Ainsworth / HXA7241",
    "date":
    [
      "2010-04-02",
      "2010-05-18"
    ],
    "description":"Definition of a minimal general data format that you can write straight into any text.",
    "language":"en-GB",
    "relation":"http://www.hxa.name/txon/",
    "subject":"data format, text, JSON, software tool, python",
    "type":"technical article"
  },
  "DC.title":"TXON: Text Object Notation -- Textual Punctuation For Data",
  "name":
  [
    "value",
    {
      "name2":"nested value"
    }
  ],
  "references":
  [
    "http://json.org/",
    "http://lua.org/"
  ]
}